homeassistant/components/cover/opengarage.py

```diff
@@ -124,19 +128,25 @@ def close_cover(self, **kwargs):
         if self._state not in [STATE_CLOSED, STATE_CLOSING]:
             self._state_before_move = self._state
             self._state = STATE_CLOSING
+            self._state_reset_count = STATE_RESET_COUNTER
             self._push_button()
 
     def open_cover(self, **kwargs):
         """Open the cover."""
         if self._state not in [STATE_OPEN, STATE_OPENING]:
             self._state_before_move = self._state
             self._state = STATE_OPENING
+            self._state_reset_count = STATE_RESET_COUNTER
             self._push_button()
 
     def update(self):
         """Get updated status from API."""
         try:
             status = self._get_status()
+            if self._state_reset_count > 0:
+                self._state_reset_count -= 1
+                if self._state_reset_count == 0:
+                    self._state_before_move = None
```

Comment:
Why do we need to keep track of this. We don't push a state update in the actuator methods (open_cover / close_cover) anyway. Why not just let the `update` method update the state to the correct state?

Reply:
Unavailable entities are still shown in the GUI. They are marked as unavailable. The state machine will store the time when the entity changed to unavailable in `last_changed` state attribute.

Reply:
@MartinHjelmare I did some testing on this. I made some code changes to set the device to Unavailable when it can't send a button press or get the state. While It doesn't disappear from the GUI, I believe the component now has less value than the current implementation which will show the user the status based on the last sensors.

I was unable to find a `last_changed` state attribute when the device went offline.

When I went into the HA templates, I entered: `{{ states.cover.testgarage }}` the output was as follows:

`<template state cover.testgarage=unavailable; friendly_name=testgarage, supported_features=3, device_class=garage @ 2018-08-01T21:22:56.414901-04:00>`

For comparison, when the device is online it looks as follows:

`<template state cover.testgarage=closed; wifi_signal=-62, distance_sensor=202, door_state=closed, friendly_name=testgarage, supported_features=3, device_class=garage @ 2018-08-01T21:50:36.429711-04:00>`

While there may be some kind of internal state change, I don't see this `last_changed` being visible.  What am I missing? Another side effect of Unavailable is that any sensors that were created from states (door status, car present, etc) are now unusable. It seems like HA removes these states.  Again, this is less than ideal.  

I mentioned before the component maintains an internal state (open, close, opening, closing). It does this for various reasons that have positive impact for the user.  I'd rather keep this functionality than change how the component works entirely.  The point of this PR was to fix a case where this internal state gets stuck. The reason for this is still unknown and hard to reproduce. All I know is it happens when HA thinks it sent the HTTP request, but it never made it to the device.

Reply:
This should show last changes attribute:
```
{{ states.cover.testgarage.last_changed }}
```